File: Code/new_fitervari_backend/src/main/java/at/htl/model/SetHistory.java
package at.htl.model;

import javax.persistence.*;

@Entity
@Table(name = "set_history")
public class SetHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private double time;

    private double distance;

    private double weight;

    private int repetitions;

    @Column(name = "set_number")
    private int setNumber;

    //region GetterSetter
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getRepetitions() {
        return repetitions;
    }

    public void setRepetitions(int repetitions) {
        this.repetitions = repetitions;
    }

    public double getTime() {
        return time;
    }

    public void setTime(double time) {
        this.time = time;
    }

    public double getDistance() {
        return distance;
    }

    public void setDistance(double distance) {
        this.distance = distance;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public int getSetNumber() {
        return setNumber;
    }

    public void setSetNumber(int setNumber) {
        this.setNumber = setNumber;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/SetHistory.kt
package at.htl_leonding.model

import io.quarkus.hibernate.orm.panache.PanacheEntity
import javax.persistence.*

@Entity
@Table(name = "set_history")
class SetHistory(
        var time: Double?,
        var distance: Double?,
        var weight: Double?,
        var repetitions: Int?,
        var exercise_history_id: Long,
        @Column(name = "set_number")
        var setNumber: Int
) : PanacheEntity() {
    fun copyValues(other: SetHistory) {
        this.time = other.time
        this.distance = other.distance
        this.weight = other.weight
        this.repetitions = other.repetitions
        this.setNumber = other.setNumber
        this.exercise_history_id = other.exercise_history_id
    }
}